test/functional/system/rounds.js

```diff
@@ -1155,4 +1182,22 @@ describe('rounds', () => {
 			return expect(lastBlock.height).to.equal(101);
 		});
 	});
+
+	describe('deleting last block of round twice in a row - no transactions during round', () => {
+		before(() => {
+			return Promise.mapSeries([...Array(202)], () => {
```

Comment:
This should be 201 since we already have genesisBlock in the database.

Reply:
:/. I don't like these tests. If in the future it fails, the person looking at the code will have to understand all the tests to see what is the state of the chain (how many blocks it contains) before he can see what the actual problem is. Besides, it makes our tests tightly coupled with each other, meaning even just changing the order will make tests fail.

Reply:
Yes, such a behavior is intended. Those tests are functional system, the longer chain is - the better reliability of those tests is.

Reply:
The purpose of these tests is not to check how the application behaves for the longer chain, they are for checking what happens when we delete the last block of the round twice. But I am not asking to change these tests, just talking about the structure we follow sometime for writing tests, so we may have a consistent style of writing tests by different developers. :)

Reply:
@SargeKhan I can refactor those tests and decouple them to few separate files (file per scenario tested), if we all agree that they will be easier to maintain then. However after `beta.7` and `beta.8` get merged together, as those branches contain separate changes to those tests.

Reply:
For simplicity of merging let's leave them in the same file for now.